openpathsampling/engines/trajectory.py

```diff
@@ -534,21 +534,27 @@ def to_mdtraj(self, topology=None):
             If not None this topology will be used to construct the mdtraj
             objects otherwise the topology object will be taken from the
             configurations in the trajectory snapshots.
-        
+
         Returns
-        -------        
+        -------
         :class:`mdtraj.Trajectory`
             the trajectory
         """
-
-
         if topology is None:
-            topology = self.topology.mdtraj
+            # TODO: maybe add better error output?
+            # if AttributeError here, engine doesAn't support mdtraj
+            topology = self[0].engine.mdtraj_topology
```

Comment:
I think this fails for Trajectories with no frames. If I remember correctly, we had a similar feature before and removed it. But I do not remember why...

Just check if length is zero and return an empty mdtraj.Trajectory should work.

Reply:
Interesting point. But I don't think that MDTraj allows empty trajectories (follow https://github.com/mdtraj/mdtraj/issues/1319 for more). For the time being, I added stuff so that we explicitly forbid conversion of a length-zero trajectory to MDTraj (raise a `ValueError` with a clear message), and a test to check it.